src/batch_write.rs

```diff
@@ -53,18 +53,25 @@ impl DbInner {
     #[allow(clippy::panic)]
     async fn write_batch(&self, batch: WriteBatch) -> Result<Arc<KVTable>, SlateDBError> {
         self.maybe_apply_backpressure().await;
+        let now = self.options.clock.now();
 
         let current_table = if self.wal_enabled() {
             let mut guard = self.state.write();
             let current_wal = guard.wal();
             for op in batch.ops {
                 match op {
-                    WriteOp::Put(key, value) => {
+                    WriteOp::Put(key, value, opts) => {
                         current_wal.put(
                             key,
                             value,
                             RowAttributes {
-                                ts: Some(self.options.clock.now()),
+                                ts: Some(now),
+                                expire_ts: opts.ttl.map(|ttl| {
```

Comment:
There are three places where this `expire_ts` is calculated in the same way. Have you considered creating a function implemented by `PutOptions` that encapsulates this logic? That might be less error prone for future usage or changes:

```rust
impl PutOptions {
  pub(crate) fn expire_ts_from(&self, now: i64) -> i64 {
    now + i64::try_from(self.ttl.as_millis()).expect(
                                         "Duration could not be converted into an i64 timestamp. \
             Perhaps the duration in millis exceeds the maximum i64?")
  }
}
```

Additional question: Should the error be handled instead of raising a panic?

Reply:
Good suggestion, I'll make the change! re: error handling, I'm not sure there's anything we can really do in this situation and I'd rather not return a `Result` in this situation since that could potentially cascade the code significantly. I don't feel too strongly about that though so if people prefer returning a result in this situation I can do that.

Reply:
> Additional question: Should the error be handled instead of raising a panic?

I mentioned this in #283, but just making a note here: #111 addresses this concern. The issue we have right now is put/delete don't return `Result`s. #111 suggests adding that and removing all panics on those call paths.